app.py

```diff
@@ -54,6 +67,21 @@ def load_post_ymls_by_tag(tag) -> dict:
 
 @dataclass
 class Post:
+
+    """Class representing a single post
+
+    Attributes:
+        content (str): HTML containing the body of the post
+        date (datetime.datetime): date the post is published
+        excerpt (str): content capped to 3000 chars
+        image (str): path to the post image
```

Comment:
IIRC, it's not really the path but rather the filename only? If it is the path, the type should be `pathlib.Path`.